### PR TITLE
refactor(launcher): Update launcher to ES6

### PR DESCRIPTION
**Pull request by lusarz**
Within this pull request I've used a few ES6 features in `lib/launcher.js` file to improve it's readability.

### CODE REVIEW

File: lib/launcher.js [hunk 1]
```diff
@@ -173,76 +134,54 @@ var Launcher = function (server, emitter, injector) {
...
+        .map((browser) => browser.forceKill())
+    ).then(callback)
   }
 
-  this.areAllCaptured = function () {
-    return !self._browsers.some(function (browser) {
-      return !browser.isCaptured()
-    })
+  this.areAllCaptured = () => {
+    return this._browsers.every((browser) => browser.isCaptured())
```
Comment (johnjbarton) [inline lib/launcher.js]: Can we use an expression return:
`() => this._browsers.every((browser) => browser.isCaptured())`
?

File: lib/launcher.js [hunk 2]
```diff
@@ -71,81 +60,53 @@ var Launcher = function (server, emitter, injector) {
...
-      self.jobs.run()
-      self._browsers.push(browser)
+      this.jobs.run()
+      this._browsers.push(browser)
     }
   }
 
-  this.launch = function (names, concurrency) {
+  this.launch = (names, concurrency) => {
+    lastStartTime = Date.now()
```
Comment (johnjbarton) [inline lib/launcher.js]: To be extra careful let's put this after new Jobs as before.

File: lib/launcher.js [hunk 3]
```diff
@@ -25,34 +26,22 @@ var baseBrowserDecoratorFactory = function (
   }
 }
 
-var Launcher = function (server, emitter, injector) {
+function Launcher (server, emitter, injector) {
```
[the remark this answers is not in the document]
Comment (johnjbarton) [reply lib/launcher.js]: (Actually I prefer small refactoring steps, so doing this next is my pref).

[the remark this answers is not in the document]
Comment (devoto13) [reply lib/launcher.js]: Okey, I see.

File: lib/launcher.js (hunk 1 above)
Comment (lusarz) [reply lib/launcher.js]: Good point, fixed!

File: lib/launcher.js (hunk 2 above)
Comment (lusarz) [reply lib/launcher.js]: Sure, fixed

**Comment by johnjbarton** [comment]
This PR now conflicts with master.

**Comment by lusarz** [comment]
Conflicts with master fixed

**Pull request merged**